src/docfx/SubCommands/DocumentBuilderWrapper.cs

```diff
@@ -83,14 +83,22 @@ public void BuildDocument()
                 {
                     BuildDocument(_config, _manager, _baseDirectory, _outputDirectory, _pluginDirectory, _templateDirectory);
                 }
-                catch (AggregateException agg) when (agg.InnerException is DocfxException || agg.InnerException is DocumentException)
+                catch (AggregateException agg) when (agg.InnerException is DocfxException)
                 {
                     throw new DocfxException(agg.InnerException.Message);
                 }
+                catch (AggregateException agg) when (agg.InnerException is DocumentException)
+                {
+                    throw;
```

Comment:
this directly throws the aggregated exception - is this what you desire?

Reply:
Yes, the outer code will look into the details: https://github.com/dotnet/docfx/pull/5634/files#diff-1413287a5e06fba6e15aab6147f23a15R95